pf/internal/schemashim/object_pseudoresource.go

```diff
@@ -32,15 +34,33 @@ import (
 // an types schema through a pseudo-Resource.
 type objectPseudoResource struct {
 	schemaOnly
-	obj types.ObjectType
+	obj          basetypes.ObjectTypable
+	nestedAttrs  map[string]pfutils.Attr
+	allAttrNames []string
+}
 
-	attrs map[string]pfutils.Attr
+func isObject(t attr.Type) (basetypes.ObjectTypable, bool) {
+	tt, ok := t.(basetypes.ObjectTypable)
+	return tt, ok
 }
 
-func newObjectPseudoResource(t types.ObjectType, attrs map[string]pfutils.Attr) *objectPseudoResource {
+func newObjectPseudoResource(t basetypes.ObjectTypable, nestedAttrs map[string]pfutils.Attr) *objectPseudoResource {
+	lowerType := t.TerraformType(context.Background())
+	objType, ok := lowerType.(tftypes.Object)
+	if !ok {
+		panic("t basetypes.ObjectTypable should produce a tftypes.Object in t.TerraformType()")
+	}
```

Comment:
Nit: `contract` is cleaner and we want to know the actual type if this code path ever triggers.
```suggestion
	contract.Assertf(ok, "t basetypes.ObjectTypable should produce a tftypes.Object in t.TerraformType(): found %T", lowerType)
```